pkg/gardenlet/controller/managedseed/managedseed_actuator.go

```diff
@@ -0,0 +1,835 @@
+	// Get gardenlet chart values
+	values, err := a.getGardenletChartValues(ctx, managedSeed, shoot)
+	if err != nil {
+		return err
+	}
+
+	// Delete gardenlet chart
+	if err := a.shootClient.ChartApplier().Delete(ctx, filepath.Join(common.ChartPath, "gardener", "gardenlet"), v1beta1constants.GardenNamespace, "gardenlet", kubernetes.Values(values)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *actuator) prepareSeedSpec(ctx context.Context, managedSeed *seedmanagementv1alpha1.ManagedSeed, shoot *gardencorev1beta1.Shoot, seedSpec *gardencorev1beta1.SeedSpec) (*gardencorev1beta1.SeedSpec, error) {
+	// Get shoot secret
+	shootSecret, err := a.getShootSecret(ctx, shoot)
+	if err != nil {
+		return nil, err
+	}
+
+	// Prepare seed backup
+	seedBackup, err := a.prepareSeedBackup(ctx, managedSeed, shoot, shootSecret, seedSpec.Backup)
+	if err != nil {
+		return nil, err
+	}
+
+	// Copy seed spec
+	spec := seedSpec.DeepCopy()
+
+	// Initialize backup
+	spec.Backup = seedBackup
+
+	// Initialize DNS
+	if spec.DNS.IngressDomain == "" && shoot.Spec.DNS != nil && shoot.Spec.DNS.Domain != nil {
+		spec.DNS.IngressDomain = fmt.Sprintf("%s.%s", common.IngressPrefix, *shoot.Spec.DNS.Domain)
+	}
+
+	// Initialize networks
+	if spec.Networks.Nodes == nil {
+		spec.Networks.Nodes = shoot.Spec.Networking.Nodes
+	}
+	if spec.Networks.Pods == "" && shoot.Spec.Networking.Pods != nil {
+		spec.Networks.Pods = *shoot.Spec.Networking.Pods
+	}
+	if spec.Networks.Services == "" && shoot.Spec.Networking.Services != nil {
+		spec.Networks.Services = *shoot.Spec.Networking.Services
+	}
+
+	// Initialize provider
+	if spec.Provider.Type == "" {
+		spec.Provider.Type = shoot.Spec.Provider.Type
+	}
+	if spec.Provider.Region == "" {
+		spec.Provider.Region = shoot.Spec.Region
+	}
+
+	// If secret reference is specified, create or update the corresponding secret
+	if spec.SecretRef != nil {
+		// Get shoot kubeconfig secret
+		shootKubeconfigSecret, err := a.getShootKubeconfigSecret(ctx, shoot)
+		if err != nil {
+			return nil, err
+		}
+
+		// Initialize seed secret data
+		data := shootSecret.Data
+		data[kubernetes.KubeConfig] = shootKubeconfigSecret.Data[kubernetes.KubeConfig]
+
+		// Create or update seed secret
+		ownerRefs := []metav1.OwnerReference{
+			*metav1.NewControllerRef(managedSeed, seedmanagementv1alpha1.SchemeGroupVersion.WithKind("ManagedSeed")),
+		}
+		if err := kutil.CreateOrUpdateSecretByRef(ctx, a.gardenClient.Client(), *spec.SecretRef, corev1.SecretTypeOpaque, data, ownerRefs); err != nil {
+			return nil, err
+		}
+	}
+
+	// Initialize settings
+	if spec.Settings == nil {
+		spec.Settings = &gardencorev1beta1.SeedSettings{}
+	}
+
+	// If settings.verticalPodAutoscaler is not specified, enable it if the shoot namespace in the seed doesn't contain
+	// a vpa-admission-controller deployment, and disable it otherwise
+	if spec.Settings.VerticalPodAutoscaler == nil {
+		disabled, err := a.disableVPA(ctx, shoot)
```

Comment:
```suggestion
		shootVPAEnabled, err := a.isVPAEnabled(ctx, shoot)
```
`disableVPA` sounds like an action.

Reply:
OK

Reply:
sounds good :)